src/java/org/apache/cassandra/db/MutationVerbHandler.java

```diff
@@ -41,6 +43,13 @@ private void failed()
 
     public void doVerb(Message<Mutation> message)
     {
+        if (approxTime.now() > message.expiresAtNanos())
```

Comment:
So the approximate times are safe to compare across nodes?

Reply:
My logic here was that:
a) approx time is _really_ bad only for metrics (but even then, only for idle/very fast clusters)
b) relative time between replicas is controlled by something like NTP anyways, and operators should know their drift
c) diff between approx and exact time is up to 2 micros, which means that for cross-node it'll be up to 4 micros; if we time out request 4 micros too early, I think it is more than fine, we should not operate that close to timeouts anyways